src/components/JaegerIntegration/JaegerResults/SpanTable.tsx

```diff
@@ -0,0 +1,217 @@
+import * as React from 'react';
+import { Span } from '../../../types/JaegerInfo';
+import { Table, TableHeader, TableBody, IRow, expandable, RowWrapperProps } from '@patternfly/react-table';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { Tooltip } from '@patternfly/react-core';
+import { KialiAppState } from '../../../store/Store';
+import { connect } from 'react-redux';
+import { formatDuration } from './transform';
+import history from '../../../app/History';
+import { Link } from 'react-router-dom';
+import { serverConfig } from '../../../config';
+import { SpanTabsTags } from './SpanTabsTags';
+import { isErrorTag } from '../RouteHelper';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/Table/table';
+import { PfColors } from '../../Pf/PfColors';
+
+interface SpanDetailProps {
+  spans: Span[];
+  namespaceSelector: boolean;
+}
+
+interface SpanDetailState {
+  spanSelected?: Span;
+  isModalOpen: boolean;
+  columns: any;
+  rows: any;
+}
+
+export class SpanTableC extends React.Component<SpanDetailProps, SpanDetailState> {
+  constructor(props: SpanDetailProps) {
+    super(props);
+    this.state = {
+      isModalOpen: false,
+      columns: [
+        {
+          title: 'Operation',
+          cellFormatters: [expandable]
+        },
+        'Service',
+        { title: 'Duration' },
+        '',
+        ''
+      ],
+      rows: this.getRows()
+    };
+  }
+
+  componentDidUpdate(prevProps: Readonly<SpanDetailProps>): void {
+    if (prevProps.spans !== this.props.spans) {
+      this.setState({ rows: this.getRows() });
+    }
+  }
+
+  goService = (service: string = this.props.spans[0].process.serviceName, extra: string = '') => {
```

Comment:
Not sure if the caller checks that this.props.spans is always populated, otherwise would send some undefined in the console. It didn't fail in the tests and this case didn't happen, but perhaps add a comment to the caller on these scenarios (for easy maintenance in future iterations).